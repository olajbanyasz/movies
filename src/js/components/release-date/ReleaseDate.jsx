import React, { Component } from 'react';
import { Badge } from 'react-bootstrap';
import './releasedate.style.less';

class ReleaseDate extends Component {
  render() {
    return (
      <Badge className='release-date'>
        {this.getReleaseYear(this.props.releaseDate)}
      </Badge>
    );
  }
      
  getReleaseYear(date) { 
    return date.split('-')[0];
  }
}

export default ReleaseDate;