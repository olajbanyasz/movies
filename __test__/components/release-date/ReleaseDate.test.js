import React from 'react';
import { shallow } from 'enzyme';
import ReleaseDate from '../../../src/js/components/release-date/ReleaseDate';

describe('ReleaseDate', () => {
  it('should rendered correctly', () => {
    const component = shallow(<ReleaseDate releaseDate={'2018-12-04'} />);
    expect(component).toMatchSnapshot();
  });
});
