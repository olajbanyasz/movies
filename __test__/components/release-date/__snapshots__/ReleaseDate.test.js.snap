// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseDate should rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReleaseDate
    releaseDate="2018-12-04"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Badge
        bsClass="badge"
        pullRight={false}
      >
        2018
      </Badge>,
      "className": "release-date",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "badge",
        "children": "2018",
        "pullRight": false,
      },
      "ref": null,
      "rendered": "2018",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Badge
          bsClass="badge"
          pullRight={false}
        >
          2018
        </Badge>,
        "className": "release-date",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "badge",
          "children": "2018",
          "pullRight": false,
        },
        "ref": null,
        "rendered": "2018",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test function on movie ReleaseDate should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReleaseDate
    releaseDate="2018-12-04"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Badge
        bsClass="badge"
        pullRight={false}
      >
        2018
      </Badge>,
      "className": "release-date",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "badge",
        "children": "2018",
        "pullRight": false,
      },
      "ref": null,
      "rendered": "2018",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Badge
          bsClass="badge"
          pullRight={false}
        >
          2018
        </Badge>,
        "className": "release-date",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "badge",
          "children": "2018",
          "pullRight": false,
        },
        "ref": null,
        "rendered": "2018",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test on ReleaseDate should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReleaseDate
    releaseDate="2018-12-04"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Badge
        bsClass="badge"
        pullRight={false}
      >
        2018
      </Badge>,
      "className": "release-date",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "badge",
        "children": "2018",
        "pullRight": false,
      },
      "ref": null,
      "rendered": "2018",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Badge
          bsClass="badge"
          pullRight={false}
        >
          2018
        </Badge>,
        "className": "release-date",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "badge",
          "children": "2018",
          "pullRight": false,
        },
        "ref": null,
        "rendered": "2018",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
